### ISSUE TITLE
Enhancement: Re-assess if the 'IsAssignableFrom' can be used over the customized 'IsInterfacedTo'.

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#469]
RepoDb property handlers are generic in which it force us to deliver a customized (almost-like) `IsAssignableFrom` equivalent method for the `System.Type`.

But since then, RepoDb has improve a lot when it comes to its abstraction and implementation. This user story will allow us to refactor the unneeded code from the code-line by re-assessing whether the `IsAssignableFrom` method can now be used as a replacement for the customized [IsInterfacedTo](https://github.com/mikependon/RepoDb/blob/e9e5e8a63065e2289984d9358e364ef384c40208/RepoDb.Core/RepoDb/Extensions/TypeExtension.cs#L66) method.

**Comment by mikependon** [comment]
Tested today, it seems invalid.

Interface:

```csharp
public interface IPropertyHandler<TInput, TOutput>
{
    TOutput Get(TInput input);
    TInput Set(TOutput input);
}
```

Class:

```csharp
public class AddressHandler : IPropertyHandler<string, object>
{
    public object Get(string input)
    {
        throw new NotImplementedException();
    }

    public string Set(object input)
    {
        throw new NotImplementedException();
    }
}
```

Main:

```csharp
class Program
{
    static void Main(string[] args)
    {
        var type = typeof(AddressHandler);
        Console.WriteLine(typeof(IPropertyHandler<,>).IsAssignableFrom(type));
        Console.WriteLine(typeof(IPropertyHandler<string, object>).IsAssignableFrom(type));
        Console.ReadLine();
    }
}
```

Output:

```
> False
> True
```

**Comment by mikependon** [comment]
This has been fixed and is now available at [RepoDb v1.12.3](https://www.nuget.org/packages/RepoDb/1.12.3). Closing this incident ticket now.
